src/client.js

```diff
@@ -23,70 +23,95 @@ const API_TYPE_CLIENT = 'client';
 const API_TYPE_APP = 'app';
 
 /**
- * Create a new StitchClient instance.
+  * Factory class to create a new StitchClient asynchronously.
+  */
+export class StitchClientFactory {
+  constructor() {
+    throw new StitchError('StitchClient can only be made from the StitchClientFactory.create function');
+  }
+
+  static create(clientAppID, options = {}) {
+    return new Promise((resolve, reject) => {
+      resolve(new StitchClient(clientAppID, options)._willInitialize);
+    });
+  }
+}
+
+/**
+ * Create a new _StitchClient instance.
```

Comment:
[nit] This comment may not apply anymore. We really do want the StitchClient documented, just not exported